chat-server/index.html

```diff
@@ -1,24 +1,123 @@
+        <label>Message: <input id="new-message-text" type="text" name="text" placeholder="The message..." /></label>
         <button type="submit">Send</button>
       </form>
-    </main>
-    <footer>
-      <h3><a href="/messages">See all messages</a></h3>
-    </footer>
-  </body>
-</html>
+    </section>
+    <section id="messages">
+      <h3><a href="" onclick="fetchAndLoadMessages()">See all messages</a></h3>
+      <div>
+        <h3>See message by ID</h3>
+        <input id="find-message-id" type="text" placeholder="Type an ID here">
+        <button onclick="fetchAndLoadMessageById()">Show message</button>
+      </div>
+      <ul id="messages-list">
+
+      </ul>
+    </section>
+  </main>
+  <footer>
+    <!-- <p>Project by Pedro Ricciardi, as part of the CYF Course - Servers Module - Chat-Server Issue</p> -->
+  </footer>
+
+  <!-- SCRIPTS -->
+  <script>
+    const renderMessage = (message) => {
+      const listItem = document.createElement('li');
+      const listItemParagraph = document.createElement('p');
+      const listItemDeleteButton = document.createElement('button');
+
+      listItemDeleteButton.textContent = '🗙';
+      listItemDeleteButton.setAttribute('class', 'delete-button');
+      listItemDeleteButton.setAttribute('onclick', 'deleteMessage(event)');
+
+      listItemParagraph.textContent = `${message.from}: ${message.text}`;
+
+      listItem.appendChild(listItemParagraph);
+      listItem.appendChild(listItemDeleteButton);
+      listItem.setAttribute('data-messageId', message.id);
+
+      return listItem;
+    };
+
+    const loadMessages = (messagesToRender) => {
+      const messagesList = document.getElementById('messages-list');
+      messagesList.innerHTML = '';
+      messagesToRender.forEach(message => {
+        const listItem = renderMessage(message);
+        messagesList.appendChild(listItem);
+      });
+    };
+
+    const loadMessagesFailure = (error) => {
+      const messagesList = document.getElementById('messages-list');
+      messagesList.innerHTML = '';
+      const errorMessage = document.createElement('p');
+      errorMessage.setAttribute('class', 'error-message');
+      errorMessage.textContent = `${error}`;
+      messagesList.appendChild(errorMessage);
+    };
+
+    const fetchAndLoadMessageById = () => {
+      const messageId = document.getElementById('find-message-id').value;
+      fetch(`/messages/${messageId}`)
+        .then(response => response.json())
+        .then(message => {
+          if (message && message.from && message.text) {
+            loadMessages([message]);
+          } else {
+            loadMessagesFailure('Message not found. Please select other ID');
+          }
+        })
+        .then(() => {
+          document.getElementById('find-message-id').value = '';
+        })
+        .catch(error => console.error('Error fetching message:', error));
```

Comment:
Will the user see this? Should they? (Same question for the other `.catch` calls below)

Reply:
Actually, I'm not sure if users need to see these errors.
Personally, I prefer not to mix scripts in the HTML file or just catch errors without handling them properly.
Since I'm usually short on time because of my job in the warehouse, I have to focus on the main goals of our tasks and sometimes skip some details.